src/addrdb.cpp

```diff
@@ -119,18 +176,49 @@ bool DeserializeFileDB(const fs::path& path, Data& data)
 }
 } // namespace
 
-CBanDB::CBanDB(fs::path ban_list_path) : m_ban_list_path(std::move(ban_list_path))
+CBanDB::CBanDB(fs::path ban_list_path)
+    : m_banlist_dat(ban_list_path.string() + ".dat"),
```

Comment:
I find `replace_extension()` a bit confusing in this case because we know that there is no extension, so we are not replacing anything (I know it will do the right thing, but still).